File: jpa-backend/src/test/java/chaincue/tech/jpabackend/masters/user_master/UserHelperTest.java
package chaincue.tech.jpabackend.masters.user_master;

import chaincue.tech.jpabackend.masters.teacher_master.TeacherHelper;
import chaincue.tech.jpabackend.utilities.ANSIColors;
import org.junit.jupiter.api.*;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;
import org.testcontainers.containers.DockerComposeContainer;
import org.testcontainers.junit.jupiter.Container;
import org.testcontainers.junit.jupiter.Testcontainers;

import java.io.File;

@SpringBootTest
//@ActiveProfiles("test")
//@Testcontainers
class UserHelperTest {

    @Container
    private static final DockerComposeContainer<?> serverContainer = new DockerComposeContainer<>(new File("src/main/resources/docker-compose-test.yml"));

    @Autowired
    TeacherHelper teacherHelper;

//    @BeforeEach
//    void setUp() {
//        serverContainer.start();
//    }
//
//    @AfterEach
//    void tearDown() {
//        serverContainer.stop();
//    }

    @Autowired
    UserHelper userHelper;

    @Test
    void saveUserSuccess() {
        String username = "user@email";
        User user = User.createUser(username);

        User result = userHelper.saveUser(user);

        ANSIColors.printBlue(result);
        Assertions.assertEquals(username, result.getUserName());
    }
    @Test
    void findUserSuccess() {
        String username = "user@email";
        var teacher = teacherHelper.saveTeacher(username);
        ANSIColors.printBlue(teacher);
        var result = userHelper.findUser(teacher.getUser().getId()).get();

        ANSIColors.printBlue(result);
        Assertions.assertEquals(username, result.getUserName());
    }

}

File: kotlin/jpa-backend/src/test/kotlin/chaincue/tech/jpabackend/masters/user_master/UserHelperTest.kt
package chaincue.tech.jpabackend.masters.user_master

import chaincue.tech.jpabackend.utilities.ANSIColors
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.testcontainers.containers.DockerComposeContainer
import org.testcontainers.junit.jupiter.Container
import java.io.File

@SpringBootTest
class UserHelperTest(@Autowired val userHelper: UserHelper) {
    companion object {
        @Container
        val serverContainer = DockerComposeContainer(File("src/main/resources/docker-compose-test.yml"))
    }

    @Test
    fun save() {
        val username = "username"
        val user = User.create(username)
        val result = userHelper.save(user)
        ANSIColors.printBlue(result)
        assertEquals(username, result.userName)
    }

    @Test
    fun findUser() {
    }
}
